src/mnt/fuse_pure.rs

```diff
@@ -138,10 +161,15 @@ fn detect_fusermount_bin() -> String {
         }
     }
     // Default to fusermount3
-    FUSERMOUNT3_BIN.to_string()
+    fusermount3_bin.to_string()
 }
 
-fn receive_fusermount_message(socket: &UnixStream) -> Result<File, Error> {
+#[cfg(target_os = "macos")]
+fn detect_fuse_program() -> String {
+    "/Library/Filesystems/macfuse.fs/Contents/Resources/mount_macfuse".to_string()
```

Comment:
On Linux we first search the PATH for `fusermount` or `fusermount3`. Is `mount_macfuse` in the PATH, and should we first use that?

Reply:
It is not on PATH unfortunately, and this is what libfuse does anyway. It takes the [full path](https://github.com/osxfuse/fuse/blob/8071bd97494cd2982ed5635525b8a754edfbfb0f/lib/fuse_darwin_private.h#L41) and [calls the fuse_resource_path function](https://github.com/osxfuse/fuse/blob/8071bd97494cd2982ed5635525b8a754edfbfb0f/lib/mount_darwin.c#L428), which in turn just [returns it as is](https://github.com/osxfuse/fuse/blob/8071bd97494cd2982ed5635525b8a754edfbfb0f/lib/fuse_darwin.c#L323-L327), since it's an absolute path.

Reply:
I see, ok. Sounds good then!